### ISSUE TITLE
Remove CTCLoss limitation of maximum target sequence length 256 (to satisify CuDNN)

### ISSUE THREAD

**Issue by KemalAltwlkany** [pytorch/pytorch#131656]
### 🚀 The feature, motivation and pitch

Hello,

I'm working on automatic speech recognition and I am experimenting with the transducer architecture. As such, I am using the [CTC loss](https://pytorch.org/docs/stable/generated/torch.nn.CTCLoss.html) .
Based on the documentation, to be CuDNN compatible, any target sequence must have a length shorter than 256.

Can this limitation be removed, as it would allow training sequence to sequence models using CTCLoss with a greater target sequence length (efficiently with CuDNN)?
